trace: atomic.AddUint64(nextSpanID) causes segmentation violation due to misalignment on 32-bit architectures

Hi, 
I am keep getting segmentation violation after updating to the latest version... It looks like the problem is here, but it could be caused by other libraries that call this one.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x804926c]

goroutine 3441 [running]:
sync/atomic.AddUint64(0x1986f66c, 0x8c3989df, 0xd1f43496, 0x6f8f1912, 0x52bbc414)
    /home/luke/bin/go/src/sync/atomic/asm_386.s:112 +0xc
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace.(*defaultIDGenerator).NewSpanID(0x1986f650, 0x0, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace/trace.go:483 +0x4c
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace.startSpanInternal(0x19d33cb0, 0x22, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace/trace.go:196 +0x95
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace.StartSpan(0x90d75c0, 0x1a0074e0, 0x19d33cb0, 0x22, 0x19a7d8dc, 0x2, 0x2, 0xfffef7ff, 0xef, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/trace/trace.go:162 +0x14c
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp.(*traceTransport).RoundTrip(0x1a0073a0, 0x19896680, 0x1a0b9580, 0x1, 0x1)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp/trace.go:58 +0xfc
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp.statsTransport.RoundTrip(0x90d11f0, 0x1a0073a0, 0x19896680, 0x19caaac8, 0x19caaac8, 0x8abcd80)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp/client_stats.go:54 +0x2ed
bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp.(*Transport).RoundTrip(0x19f2ca40, 0x19896600, 0x19896600, 0x0, 0x8057868)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/go.opencensus.io/plugin/ochttp/client.go:89 +0x17e
bitbucket.org/xxx/xxx/vendor/golang.org/x/oauth2.(*Transport).RoundTrip(0x19f2cbc0, 0x19f5c200, 0x0, 0x0, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/golang.org/x/oauth2/transport.go:56 +0xd4
net/http.send(0x19f5c200, 0x90d1290, 0x19f2cbc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x19f5c200, 0x19caae08, ...)
    /home/luke/bin/go/src/net/http/client.go:252 +0x151
net/http.(*Client).send(0x19f2cbe0, 0x19f5c200, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2, 0xb76fbf5b, 0xb76fbf02, ...)
    /home/luke/bin/go/src/net/http/client.go:176 +0xaa
net/http.(*Client).Do(0x19f2cbe0, 0x19f5c200, 0x19828160, 0x19f5c200, 0x0)
    /home/luke/bin/go/src/net/http/client.go:615 +0x23c
bitbucket.org/xxx/xxx/vendor/golang.org/x/net/context/ctxhttp.Do(0x90d7580, 0x19828160, 0x19f2cbe0, 0x19f5c180, 0x0, 0x0, 0x19f1f864)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/golang.org/x/net/context/ctxhttp/ctxhttp.go:30 +0x59
bitbucket.org/xxx/xxx/vendor/google.golang.org/api/gensupport.SendRequest(0x90d7580, 0x19828160, 0x19f2cbe0, 0x19f5c180, 0x19a7ddbc, 0x19f2cdc0, 0x19f5c180)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/google.golang.org/api/gensupport/send.go:53 +0x142
bitbucket.org/xxx/xxx/vendor/google.golang.org/api/storage/v1.(*ObjectsInsertCall).doRequest(0x19d31560, 0x8b43385, 0x4, 0x0, 0x0, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/google.golang.org/api/storage/v1/storage-gen.go:8948 +0x490
bitbucket.org/xxx/xxx/vendor/google.golang.org/api/storage/v1.(*ObjectsInsertCall).Do(0x19d31560, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/google.golang.org/api/storage/v1/storage-gen.go:8960 +0x76
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage.(*Writer).open.func1.2(0x0, 0x80577ec)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage/writer.go:156 +0x3f
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage.runWithRetry.func1(0x19a8f240, 0x19b2a588, 0x8057868)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage/invoke.go:27 +0x1c
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/internal.retry(0x90d7580, 0x19828160, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/internal/retry.go:39 +0x37
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/internal.Retry(0x90d7580, 0x19828160, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/internal/retry.go:32 +0x48
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage.runWithRetry(0x90d7580, 0x19828160, 0x19ada7c8, 0xffffffff, 0x0)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage/invoke.go:26 +0x52
bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage.(*Writer).open.func1(0x1a08a5a0, 0x1a0961e0, 0x19caac08, 0x19caac18, 0x1, 0x1)
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage/writer.go:154 +0x58b
created by bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage.(*Writer).open
    /home/luke/golang/src/bitbucket.org/xxx/xxx/vendor/cloud.google.com/go/storage/writer.go:108 +0x3e6`
```

--- a/trace/trace.go
+++ b/trace/trace.go
@@ -470,10 +470,20 @@ func init() {
 
 type defaultIDGenerator struct {
 	sync.Mutex
-	traceIDRand *rand.Rand
+
+	// Please keep these as the first fields
+	// so that these 8 byte fields will be aligned on addresses
+	// divisible by 8, on both 32-bit and 64-bit machines when
+	// performing atomic increments and accesses.
+	// See:
+	// * https://github.com/census-instrumentation/opencensus-go/issues/587
+	// * https://github.com/census-instrumentation/opencensus-go/issues/865
+	// * https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	nextSpanID uint64
+	spanIDInc  uint64
+
 	traceIDAdd  [2]uint64
-	nextSpanID  uint64
-	spanIDInc   uint64
+	traceIDRand *rand.Rand
 }
 
 // NewSpanID returns a non-zero span ID from a randomly-chosen sequence.

